### ISSUE TITLE
Paused jobs should not have builds come back after they're aborted

### ISSUE THREAD

**Issue by danger-ranger** [concourse/concourse#482]
Not much more to add. 

I've got a bunch of paused jobs that are triggered by 'put'ting a timestamp passing a previous job. After said previous job finished the paused job was triggered but, as expected, remains pending.

But every time I cancel it it comes back again. On six identical pipelines.

Probably not a big deal, but.

**Comment by xtreme-gavin-enns** [comment]
P.S. v1.3.0

**Comment by vito** [comment]
I don't think we'll fix this. A job being paused just means its builds won't run until you unpause it. Builds will still be scheduled as normal. This way you can know that upon unpausing a job a build will run.

Can you explain your usage pattern some more? I can see how it would get annoying in some cases.

**Comment by xtreme-gavin-enns** [comment]
This was in an environment redeploy pipeline (single environment, no funny business with pools)

It cycles an environment, first destroying and then deploying. 

We wanted to tear down all of our environments, so we paused the install_opsman jobs (on each pipeline) and triggered the destroy jobs. 

Since we didn't intend to spin them back up again soon it was just annoying to have the pending animation on the paused jobs for weeks. And we didn't want to mess around with trying to unpause and then cancel before any aws resources got created (which would have worked fine).

Like I said, not a big deal.

I think this is probably just another case of concourse-as-ci not mapping well to our use case of automating shared resource maintenance. I also haven't ruled out that there is just something we haven't grokked yet :)

**Comment by Zteve** [comment]
I agree that the 'pending' animation is annoying on a paused job, and the log says it is "checking that the job is not paused", which implies it is actually spinning :-( and not just dormant.

Furthermore, the job id changes to brown (aborted) colour in the log when you cancel it (X) and then reverts after going back to pipeline view. This implies that something has happened. If the action is not allowed, it would be nice if the i/f said so.

**Comment by vito** [comment]
the build changes to brown because it was aborted. in the pipeline view, the fact that the job is paused takes precedence over all states, not just aborted.

i don't find the animation annoying on paused jobs; it's useful to know that something is waiting for it to be unpaused and that something is going to immediately happen once it's unpaused.

a job being paused is an irregular, temporary state; things will look a little funny.

**Comment by xtreme-gavin-enns** [comment]
I think it's more of a problem that you can't actually cancel a paused job. Which means that you have to move quickly when you unpause to stop a job from executing, which is problematic when the concourse web ui is being slow.

**Comment by xtreme-gavin-enns** [comment]
Then again, trying to manipulate a concourse pipeline in this way feels like using the wrong tool for the job, which begs the question of whether concourse even wants to support that kind of use case.

**Comment by vito** [comment]
Just to dig a bit further: I think the behavior of pausing jobs today works pretty much exactly as I'd expect to use it. However there may be some *other* thing that you're trying to do, that Concourse can't properly express right now, but "pausing jobs" gives you 90% of. For example: is it more accurate to say that you want to prevent new versions of a particular resource going to a job?

**Comment by Zteve** [comment]
I guess that if a job is paused (but not triggered) it is unanimated. When it is triggered (manually or automatically) then it becomes animated, and as soon as it is unpaused something will (immediately) happen.

I would like the option of cancelling the pending job caused by the trigger while the job is (and remains) paused.

I expected "cancel" to do exactly that, but it does nothing.  If you allowed it to "untrigger" the job, that would be the behaviour I would expect, and I would expect the job to stop flashing/glowing/pulsing/wotever then.

It doesn't seem (to me) to be hurting anyone to allow the "cancel" to do this, since it doesn't appear to do anything (at all?/sensible) at the moment. WDYT?

**Comment by vito** [comment]
You sure you can't cancel them?

![optimised](https://cloud.githubusercontent.com/assets/1880/17526282/6d1c6c12-5e1b-11e6-94ee-670a34ebe7fa.gif)

**Comment by Zteve** [comment]
Well, I tried hard on `cf-java-client`, and _temporarily_ I see the pending glimmers go, but they come back pretty quick.  So YES I'm sure I can't cancel them.  Go back to your demo and check it doesn't revert. Or else go try it on `cf-java-client` (Integration Test job).

And if yours doesn't revert I'm accusing you of cheating.  😄

**Comment by xtreme-gavin-enns** [comment]
Alex, maybe it behaves the way you're seeing when triggered manually, but the way we're seeing when triggered by a new version?

**Comment by xtreme-gavin-enns** [comment]
In my use case it was a paused job triggered by a resource passing a previous job that couldn't be canceled. Unfortunately those vms have all been terminated so I can't quickly reproduce it, but the setup is well documented/automated so it wouldn't be difficult.

**Comment by vito** [comment]
Ah, yeah, new builds will be enqueued if there are new versions available, and aborting will not prevent it from being enqueued as the logic for enqueueing and determining the versions is decoupled (Concourse doesn't lock down the queued build's inputs until it runs).

**Comment by chendrix** [comment]
@anEXPer came to us with a scenario where this bug interacts with https://github.com/concourse/concourse/issues/520#issuecomment-231260929 to cause utter chaos.

**Comment by anEXPer** [comment]
This is profoundly infuriating in connection to https://github.com/concourse/concourse/issues/520. If you get several triggered jobs pending (and thus blocking each other) and can't cancel them fast enough because they keep triggering, you will want to bomb your CI instance.

**Comment by vito** [comment]
Once #3857 is done we can change the scheduler to still mark a job as 'has new inputs' without queuing a pending build. Then we still have the UI feedback without the build having to be created.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.

**Comment by taylorsilva** [comment]
This hasn't been an issue since the scheduling changes made in v6.0.0. If jobs/pipelines are paused, we don't schedule builds for them.
